In [ ]:
# 拉取代码库和下载python依赖
!git clone https://github.com/i2w3/pypy
!pip install numpy
!pip install pandas
!pip install pillow
!pip install scikit-learn
!pip install opencv-python

Cloning into 'pypy'...
remote: Enumerating objects: 4068, done.
remote: Counting objects: 100% (4068/4068), done.
remote: Compressing objects: 100% (812/812), done.
remote: Total 4068 (delta 3272), reused 4038 (delta 3250), pack-reused 0
Receiving objects: 100% (4068/4068), 22.23 MiB | 27.96 MiB/s, done.
Resolving deltas: 100% (3272/3272), done.


colab不断连代码，按F12进入网站开发者选项，在console输入以下代码
```console
function ConnectButton(){
    console.log("Connect pushed"); 
    document.querySelector("#connect").click() 
}
setInterval(ConnectButton,60000);
```

In [4]:
%cd pypy
!ls
!python PCAcv.py

/content/pypy
Bayes.py	 DataProcess.py  main.py   pdData.csv  人脸图像识别项目说明.doc
build_pdData.py  face		 PCAcv.py  README.md
2412 已经从 512 * 512 压缩到 128 * 128
2416 已经从 512 * 512 压缩到 128 * 128
/usr/local/lib/python3.7/dist-packages/joblib/externals/loky/process_executor.py:691: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.
  "timeout or by a memory leak.", UserWarning
GaussianNB Best parameter (CV score=0.549):
{'pca__n_components': 73, 'pca__svd_solver': 'auto', 'pca__whiten': True}
/usr/local/lib/python3.7/dist-packages/joblib/externals/loky/process_executor.py:691: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.
  "timeout or by a memory leak.", UserWarning
BernoulliNB Best parameter (CV score=0.670):
{'pca__n_components': 116, 'pca__svd_solver': 'auto', 'pca__whiten': True}


从上面结果可以看出：Gaussian Naive Bayes分类器对应的PCA保留特征数量为73个左右较佳；Bernoulli Naive Bayes分类器对应的PCA保留特征数量为116个做优较佳

（以上结果均是在Google云代码平台上运行的，大约需要为5小时）